### DISCUSSION TITLE
other shapes of node

### DISCUSSION THREAD

**Discussion by VahidFiroozian** [bcakmakoglu/vue-flow#1023] [Q&A] [answered]
how can we use other shapes of nodes like rhombus and parallelogram node?

**Comment by bcakmakoglu** [comment, marked as answer]
As that is part of a paid example on React Flow, I won't really provide any code for this, sorry.

I can give you an idea though.
1. Create a custom node
2. Use an svg of your shape

That's it.

**Reply by VahidFiroozian** [reply]
thanks for your answer and your good library

**Comment by kvedula18** [comment]
@bcakmakoglu Thank you very much for an excellent library. We're evaluating using this library as part of our product offering. I was wondering if you are open to providing support/customization in this context, providing custom shapes like in ReactFlow Pro.

**Reply by bcakmakoglu** [reply]
Tbh, custom "shapes" are really not that hard to implement 😄 Don't overthink it.
As I said, it's just a custom node with an svg image as the "shape" and that's it, there's no magic to it whatsoever

**Reply by kvedula18** [reply]
I was able to implement custom shapes using SVG and style them based on events (selected etc). I have extended to include data attributes to store the question on the node, and conditions on the edges etc.

What I discovered is that if I use bgColor, then the pattern is ignored, however, if I use a style for the bgColor, then the pattern is honored. 

It would have been nice to include these type of examples, But, as you pointed out, these are well within the framework's capabilities.

Thank you once again for a great library.


<img width="1062" alt="image" src="https://github.com/bcakmakoglu/vue-flow/assets/45952128/a1787e9b-7c87-4a01-88b2-5adf9b20da1e">

**Reply by JenuelDev** [reply]
great work mate, is it possible if you could share to us how you did it? maybe in a blog or a youtube video? that would be nise. :)

**Comment by Xsmael** [comment]
I have a question, 
i've discovered this library first, since i use vueJS and not react. But I realised it was heaviliy inspired from reactflow. Since this is an open source project and anyone can contribute, is it forbidden to implement features that are considred PRO in reactflow or something similar and make PR ?  and if someone does it without even knowing the existance of reactflow and submits a PR, will you reject it ?
Thanks!

**Reply by bcakmakoglu** [reply]
There are no "pro" features in React Flow, a pro subscription only gives you access to specific examples that are a little more complex than the free ones.

The only thing I will not add to Vue Flow is examples that are behind the paywall in React Flow, other than that the library has all the same features as React Flow :)

**Reply by Xsmael** [reply]
Alright, so if my understanding is right, all the specific examples available in the React Flow pro subscription can be done using Vue Flow, but everyone have to do it themselves

**Reply by bcakmakoglu** [reply]
Exactly 😄 
I can give you pointer as to the direction you should take but I cannot provide you code or anything that would be similar to what is offered behind their paywall.
